### ISSUE TITLE
LocateAllOnScreen Function gives me not all images on Screen

### ISSUE THREAD

**Issue by Yavuz93** [asweigart/pyautogui#207]
Hey, 
i'm trying to read a Sudoku Board with PyAutoGUI.

Here is an example image for the Number 1 and the Grid im searching in. 
![1](https://user-images.githubusercontent.com/17688995/34566547-43d9001c-f15f-11e7-8621-583e1d7e1be2.PNG)
![grid](https://user-images.githubusercontent.com/17688995/34566548-44183e08-f15f-11e7-92bd-39af5a869882.PNG)

I have done it for every number and sometimes i get multiple answers and sometimes just one. 
Is there a way to reduce the accuracy so i can find all numbers on screen? (Or any other way)

**Comment by Yavuz93** [comment]
For better understanding i can show some more examples of grids.

With the function i found only one "Number 1" on the screen:
![1time](https://user-images.githubusercontent.com/17688995/34569079-f5012c86-f167-11e7-990c-35614b272b5b.PNG)

But on this image i can find the "Number 1" two times (still not all parts where the Number 1 is):
![2times](https://user-images.githubusercontent.com/17688995/34569080-f573c7e6-f167-11e7-849c-cbd29eb8f65c.PNG)

Here i can get 3 times the Number 1:
![3times](https://user-images.githubusercontent.com/17688995/34569081-f59a3e62-f167-11e7-8c2b-034a826d9f0a.PNG)


Why is it so random and why i can't find all numbers? I tried it with all numbers and many different snippets for the numbers. 

The only thing i can imagine is that they are not Pixelperfect, but sometimes it works for multiple Numbers and sometimes just for one...

**Comment by devnowcommit** [comment]
Any news on this? Because I have High Sierra and it doesn't work for me. It doesn't find any button.png. I've even tried the check retina display patch but stil nothing. 

Any news? Do I have to install any other packages with pip?

**Comment by Hexen-Leszy** [comment]
https://stackoverflow.com/questions/32907314/python-how-can-i-find-an-image-on-screen-by-using-pyautogui-lib

For me worked with *rbg solution

**Comment by vikramdurai** [comment]
@Yavuz93 have you checked above solution (@Hexen-Leszy), if it works close the issue.

**Comment by Yavuz93** [comment]
Yes i checked it, but it still does not work for me.

**Comment by Yavuz93** [comment]
@Hexen-Leszy Could you please show me your code if it worked for you.. import pyautogui
from PIL import Image

`
import pyautogui
from PIL import Image

Image.open("1.png").convert("RGB").save("1.png")
Image.open("grid.png").convert("RGB").save("grid.png")

i = 0
for pos in pyautogui.locateAllOnScreen('1.png'):
    print(i, pos)
    i+=1
`

**Comment by Hexen-Leszy** [comment]
Sadly I cannot show you my code ( company policy). Try make also screenshot  pyautogui.

**Comment by mogamepro** [comment]
I realize this is a few months old but in case an answer is still needed (or it helps anyone coming here and seeing this):
Try the confidence keyword. if you follow pyautogui.locateAllOnScreen down a couple levels you find that it uses pyscreeze's `_locateAll_opencv` (if you have opencv installed). It has a confidence parameter that defaults to 0.999. Just lower it until it reliably finds all the numbers you're wanting without false matches. So lower it slowly. If you don't have opencv installed, then it uses `_locateAll_python` and the confidence parameter is ignored and won't help.

You can also use `grayscale=True` but that's more of a help when dealing with colors, not the black/white of your images.

So basically it would look like this:
`pyautogui.locateAllOnScreen(image, grayscale=True, confidence=0.95)`

The docs for pyautogui is a bit lacking in some aspects and pyautogui has capabilities not listed there. 

https://github.com/asweigart/pyscreeze/blob/3da0575b746f38dcab0ff4d82d7aa280948aff58/pyscreeze/__init__.py#L106
